server/series/models.py

```diff
@@ -0,0 +1,68 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django_prometheus.models import ExportModelOperationsMixin
+
+from server.core.models import Player, Team
+from server.utils import default_invitation_expiry_date
+
+
+class Season(ExportModelOperationsMixin("season"), models.Model):  # type: ignore[misc]
+    start_date = models.DateField()
+    end_date = models.DateField()
+    name = models.CharField(max_length=255)
+
+
+class Series(ExportModelOperationsMixin("series"), models.Model):  # type: ignore[misc]
+    class Type(models.TextChoices):
+        MIXED = "MXD", _("Mixed")
+        OPENS = "OPN", _("Opens")
+        WOMENS = "WMN", _("Womens")
+
+    class Category(models.TextChoices):
+        CLUB = "Club", _("Club")
+        COLLEGE = "College", _("College")
+        SCHOOL = "School", _("School")
+        STATE = "State", _("State")
+
+    start_date = models.DateField()
+    end_date = models.DateField()
+    name = models.CharField(max_length=255)
+    type = models.CharField(max_length=3, choices=Type.choices)
+    catgory = models.CharField(max_length=7, choices=Category.choices)
+    series_roster_max_players = models.PositiveSmallIntegerField()
+    event_min_players_male = models.PositiveSmallIntegerField()
+    event_min_players_female = models.PositiveSmallIntegerField()
+    event_max_players_male = models.PositiveSmallIntegerField()
+    event_max_players_female = models.PositiveSmallIntegerField()
+    player_transfer_window_start_date = models.DateField(blank=True, null=True)
+    player_transfer_window_end_date = models.DateField(blank=True, null=True)
+    season = models.ForeignKey(Season, on_delete=models.SET_NULL, null=True, blank=True)
+    teams = models.ManyToManyField(Team, related_name="series")
+
+
+class SeriesRegistration(ExportModelOperationsMixin("series_registration"), models.Model):  # type: ignore[misc]
+    series = models.ForeignKey(Series, on_delete=models.CASCADE)
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    player = models.ForeignKey(Player, on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ["series", "team", "player"]
+
+
+class SeriesRosterInvitation(ExportModelOperationsMixin("series_roster_invitation"), models.Model):  # type: ignore[misc]
+    class Status(models.TextChoices):
+        PENDING = "Pending", _("Pending")
+        ACCEPTED = "Accepted", _("Accepted")
+        DECLINED = "Declined", _("Declined")
+        EXPIRED = "Expired", _("Expired")
+
+    series = models.ForeignKey(Series, on_delete=models.CASCADE)
+    from_player = models.ForeignKey(
+        Player, on_delete=models.CASCADE, related_name="series_invitations_sent"
```

Comment:
on delete we should be careful.. better would be do nothing?

Reply:
hmm. we could leave it be i guess, we wouldn't know who sent the invitation, if thats fine

Reply:
ok this is just invitation.. so cascade is fine.. For proper registration we should be careful

Reply:
yeah, thats why i left. will not do this for registration